Civi/Mailbatch/MailUtils.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Civi\Mailbatch;
+
+class MailUtils {
+
+  /**
+   * Get a list of the available/allowed sender email addresses
+   */
+  public static function getSenderOptions($asHtml = TRUE) {
+    // TODO: Remove check when minimum core version requirement is >= 6.0.0.
+    if (class_exists('\Civi\Api4\SiteEmailAddress')) {
+      $from_email_addresses = \Civi\Api4\SiteEmailAddress::get(FALSE)
+        ->addSelect('display_name', 'email', 'id')
+        ->addWhere('domain_id', '=', 'current_domain')
+        ->addWhere('is_active', '=', TRUE)
+        ->addOrderBy('is_default', 'DESC')
+        ->execute()
+        ->indexBy('id')
+        ->getArrayCopy();
+      // Include "email" column as the option value label did.
+      $from_email_addresses = array_map(
+        fn($address) => sprintf(
```

Comment:
Below `htmlspecialchars()` is used. Isn't it necessary here?

Reply:
`htmlspecialchars()` is applied on the result of this, so this should be fine …?

Reply:
You're right. Seems that I wasn't mindful enough, when reading the code.